File: api/models/persona.py
"""Persona model — a hypothetical employee profile for simulation."""

from uuid import UUID, uuid4

from pydantic import BaseModel, Field

from api.models.asset_allocation import AssetAllocation


class Persona(BaseModel):
    """A hypothetical employee profile for simulation."""

    id: UUID = Field(default_factory=uuid4)
    name: str
    label: str
    age: int = Field(ge=18, le=80)
    tenure_years: int = Field(ge=0, le=60)
    salary: float = Field(ge=0)
    deferral_rate: float = Field(ge=0.0, le=1.0)
    current_balance: float = Field(ge=0)
    allocation: AssetAllocation
    include_social_security: bool = True
    ss_claiming_age: int = Field(default=67, ge=62, le=70)
    hidden: bool = False

File: app/src/types/persona.ts
export interface TargetDateAllocation {
  type: 'target_date'
  target_date_vintage: number
}

export interface CustomAllocation {
  type: 'custom'
  stock_pct: number
  bond_pct: number
  cash_pct: number
}

export type AssetAllocation = TargetDateAllocation | CustomAllocation

export interface Persona {
  id: string
  name: string
  label: string
  age: number
  tenure_years: number
  salary: number
  deferral_rate: number
  current_balance: number
  allocation: AssetAllocation
  include_social_security: boolean
  ss_claiming_age: number
  hidden: boolean
}
